packages/web-components/fast-foundation/src/avatar/avatar.ts

```diff
@@ -0,0 +1,99 @@
+import { attr, FASTElement } from "@microsoft/fast-element";
+
+type AvatarShape = "circle" | "square"
+
+/**
+ * An Avatar Custom HTML Element
+ * 
+ * @public
+ */
+export class Avatar extends FASTElement {
+  /**
+   * Indicates the Avatar should have a name.
+   * 
+   * @public
+   * @remarks
+   * HTML Attribute: name
+   */
+  @attr public name: string;
+
+  /**
+   * Indicates the Avatar should have initials.
+   * 
+   * @public
+   * @remarks
+   * HTML Attribute: initials
+   */
+  @attr public initials: string;
+
+  /**
+   * Invoked when the `name` property changes.
+   * 
+   * @public
+   * @remarks
+   */
+  protected nameChanged(): void {
+    const result = this.name?.match(/\b\w/g) || [];
```

Comment:
more readable if propertyChanged() function immediately follows the property it is associated with